packages/dataviews/src/types.ts

```diff
@@ -191,7 +191,33 @@ export interface ViewList extends ViewBase {
 	};
 }
 
-export type View = ViewList | ViewBase;
+export interface ViewGrid extends ViewBase {
+	type: 'grid';
+
+	layout: {
+		/**
+		 * The field to use as the primary field.
+		 */
+		primaryField: string;
+
+		/**
+		 * The field to use as the media field.
+		 */
+		mediaField: string;
+
+		/**
+		 * The fields to use as columns.
+		 */
+		columnFields: string[];
```

Comment:
**This is strictly informational - no changes requested.**

I like to use the readonly types where they make sense, for example:

```suggestion
		columnFields: ReadonlyArray< string >;
```

It's largely the same, except it clearly indicates the array isn't intended to be modified, so things like `columnFields.pop()` wouldn't be allowed.

I don't know how much that makes sense here, but much of our code avoids mutation and this is a way to clearly express (and enforce!) that.

Note that it's probably most helpful on function parameters (or constants):

```ts
// This function accepts regular and readonly arrays. Regular arrays satisfy readonly.
// it's easy to use and we know it won't modify the array internally
function sum(ns: ReadonlyArray<number>): number { return ns.reduce((a, b) => a + b, 0); }

// This function is trickier, we probably shouldn't care about returning readonly.
// Callers should be able to do what they want!
function split(s: string): ReadonlyArray<string> { return s.split(''); }
```

Reply:
For me I'd prefer to avoid this here because it will create inconsistencies. Most of the types are should be "Readonly" types.